packages/indexer-agent/src/agent.ts

```diff
@@ -392,6 +396,13 @@ class Agent {
       }) => {
         this.logger.info(`Reconcile with the network`)
 
+        // should indexer have the ability to use a longer lifetime than network max?
+        // or use min like `? Math.min(this.indexer.maxLifetime, maxAllocationEpochs)`
+        // agent can still serve queries without rewards
+        const indexerMaxAllocationEpochs = this.indexer.maxLifetime
+          ? this.indexer.maxLifetime
+          : maxAllocationEpochs
```

Comment:
currently, indexer have the ability to use a longer lifetime than network specified, since agent can still serve queries without rewards? or we can take the min `? Math.min(this.indexer.maxLifetime, maxAllocationEpochs)`